allensdk/test/brain_observatory/test_session_analysis_regression.py

```diff
@@ -0,0 +1,225 @@
+
+@pytest.fixture(scope="module")
+def analysis_a_new(nwb_a):
+    with tempfile.NamedTemporaryFile(delete=True) as tf:
+        save_path = tf.name
+
+    logging.debug("running analysis a")
+    session_analysis = SessionAnalysis(nwb_a, save_path)
+    session_analysis.session_a(plot_flag=False, save_flag=True)    
+    logging.debug("done running analysis a")
+    logging.debug(save_path)
+
+    yield save_path
+
+    if os.path.exists(save_path):
+        os.remove(save_path)
+
+@pytest.fixture(scope="module")
+def analysis_b_new(nwb_b):
+    with tempfile.NamedTemporaryFile(delete=True) as tf:
+        save_path = tf.name
+
+    logging.debug("running analysis b")
+    session_analysis = SessionAnalysis(nwb_b, save_path)
+    session_analysis.session_b(plot_flag=False, save_flag=True)    
+    logging.debug("done running analysis b")
+    logging.debug(save_path)
+
+    yield save_path
+
+    if os.path.exists(save_path):
+        os.remove(save_path)
+
+@pytest.fixture(scope="module")
+def analysis_c_new(nwb_c):
+    with tempfile.NamedTemporaryFile(delete=True) as tf:
+        save_path = tf.name
+
+    logging.debug("running analysis c")
+    session_analysis = SessionAnalysis(nwb_c, save_path)
+    session_analysis.session_c(plot_flag=False, save_flag=True)    
+    logging.debug("done running analysis c")
+    logging.debug(save_path)
+
+    yield save_path
+
+    if os.path.exists(save_path):
+        os.remove(save_path)
+
+
+@pytest.mark.skipif(os.getenv('TEST_COMPLETE') != 'true',
+                    reason="partial testing")
+def test_session_a(dg, nm1a, nm3, nwb_a, analysis_a, analysis_a_new):
+    peak = pd.read_hdf(analysis_a, "analysis/peak")
+    new_peak = pd.read_hdf(analysis_a_new, "analysis/peak")
+    assert np.allclose(peak, new_peak)
+
+    print("reading outputs")
+    dg_new = DriftingGratings.from_analysis_file(BODS(nwb_a), analysis_a_new)
+    #assert np.allclose(dg.sweep_response, dg_new.sweep_response)
+    assert np.allclose(dg.mean_sweep_response, dg_new.mean_sweep_response)
+        
+    assert np.allclose(dg.response, dg_new.response)
+    assert np.allclose(dg.noise_correlation, dg_new.noise_correlation)
+    assert np.allclose(dg.signal_correlation, dg_new.signal_correlation)
+    assert np.allclose(dg.representational_similarity, dg_new.representational_similarity)
+    
+    nm1a_new = NaturalMovie.from_analysis_file(BODS(nwb_a), analysis_a_new, si.NATURAL_MOVIE_ONE)
+    #assert np.allclose(nm1a.sweep_response, nm1a_new.sweep_response)
+    assert np.allclose(nm1a.binned_cells_sp, nm1a_new.binned_cells_sp)
+    assert np.allclose(nm1a.binned_cells_vis, nm1a_new.binned_cells_vis)
+    assert np.allclose(nm1a.binned_dx_sp, nm1a_new.binned_dx_sp)
+    assert np.allclose(nm1a.binned_dx_vis, nm1a_new.binned_dx_vis)
+    
+    nm3_new = NaturalMovie.from_analysis_file(BODS(nwb_a), analysis_a_new, si.NATURAL_MOVIE_THREE)
+    #assert np.allclose(nm3.sweep_response, nm3_new.sweep_response)
```

Comment:
Is there anything to check here?

Reply:
sweep response is just a shuffle of the dff traces in the nwb file into a dataframe, so I'm not going to check those.  We don't compute mean_sweep_response for natural movies, so nothing to check.  I will remove this.